eos-paygd/main.c

```diff
@@ -268,6 +271,76 @@ test_and_update_securitylevel (void)
   return TRUE;
 }
 
+static gboolean
+open_log_file (GIOChannel **log_io)
+{
+  int system_ret;
+  g_autoptr(GFile) log_file = NULL;
+  g_autoptr(GFileInfo) log_file_info = NULL;
+  g_autoptr(GError) error = NULL;
+  const gchar *log_file_name = NULL;
+
+  /* Use /bin/mkdir instead of mkdir() to ensure the mode is unaffected by
+   * the process's umask.
+   */
+  system_ret = system ("/bin/mkdir -p --mode=755 " LOGFILE_DIRNAME);
```

Comment:
Can we create the directory with `systemd-tmpfiles` instead?

Reply:
Or, looks like we can use `LogsDirectory=eos-paygd` in the unit file to cause systemd to create it before launching eos-paygd.

Reply:
I don't think so, as `systemd-tmpfiles` would not have run before paygd starts, and `LogsDirectory=` would add a `RequiresMountsFor=/var` dependency which I believe would either delay paygd's startup or create an impossible transaction.

I'm using `mkdir` following how `/var/lib/eos-payg` is created a few lines below.